rero_ils/modules/ill_requests/views.py

```diff
@@ -78,3 +90,57 @@ def get_patron(loc_pid):
 
     return render_template('rero_ils/ill_request_form.html',
                            form=form, viewcode=viewcode)
+
+
+def _populate_document_data_form(doc_pid, form):
+    """Populate document data in the form.
+
+    :param: doc_pid: Document PID.
+    :param: form: The ill form.
+    """
+    if not (doc := Document.get_record_by_pid(doc_pid)):
+        return
+    # Document title
+    form.document.title.data = create_title_text(doc.get('title'))
+    # Document authors (only first three)
+    statements = doc.get('responsibilityStatement', [])
+    authors = []
+    for statement in statements:
+        authors.extend(author.get('value') for author in statement)
+    if authors:
+        form.document.authors.data = '; '.join(authors[:3])
+    # Document publisher and year
+    productionActivity = doc.get('provisionActivity', [])
+    publishers = []
+    dates = []
+    for provision in productionActivity:
+        if date := provision.get('startDate'):
+            dates.append(date)
+        if statements := provision.get('statement'):
+            publishers.extend(statement.get('label')[0].get('value')
+                              for statement in statements
+                              if (statement.get('type') == 'bf:Agent')
+                              and statement.get('label', []))
+    # Document publisher
+    if publishers:
+        form.document.publisher.data = publishers[0]
+    # Document date
+    if dates:
+        form.document.year.data = dates[0]
+    # Document identifier
+    if identifiedBy := list(filter(lambda i: i.get('type') in [
+            'bf:Ean', 'bf:Isbn', 'bf:Issn', 'bf:IssnL'],
+                                doc.get('identifiedBy'))):
+        indentifed = identifiedBy[0]
+        type = _(indentifed.get('type'))
+        value = indentifed.get('value')
+        form.document.identifier.data = f'{value} ({type})'
```

Comment:
I create a set of method to deal with document identifiers.
My suggestion was to create a "get_document_identifiers(document: dict, types: list[str])" into `documents.utils`.
This method take as first parameter the `document`, and as second parameters a list of 'identifierType' to return.
```python
def get_document_identifiers(doc, types=None):
    assert types  # ensure a least one type is asked
    for identifier in doc.get('identifiedBy', []):
        if identifier['type'] in types:
            yield IdentifierFactory.create_identifier(identifier)
```
you could call it 
```pyton
types = [IdentifierType.ISBN, IdentifierType.EAN, IdentifierType.ISSN, IdentifierType.L_ISSN]
if identifiers := next(iter(get_document_identifiers(doc, types), None)):
    form.document.identifier.data =
        f'{identifier.normalize()} ({_(identifier.type)}'
```

Reply:
It should be better to use an `IdentifierRenderer` subclass but it seems a little bit complicated to translate the identifier type (because this rendered should work only with the app context...)